SeleniumWebDriverBasics/YandexUnitTests.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Emit;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Firefox;
+using OpenQA.Selenium.Support.UI;
+using static System.Net.WebRequestMethods;
+
+namespace SeleniumWebDriverBasics
+{
+    [TestFixture]
+    public class YandexUnitTests
+    {
+        private IWebDriver driver;
+        private string baseUrl;
+
+        [SetUp]
+        public void TestSetup()
+        {
+            var service = FirefoxDriverService.CreateDefaultService();
+            this.driver = new FirefoxDriver(service);
+            this.driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(10);
+            this.baseUrl = "https://mail.yandex.com/";
+            this.driver.Navigate().GoToUrl(this.baseUrl);
+            this.driver.Manage().Window.Maximize();
+        }
+
+        [Test]
+        public void LoginToYandex()
+        {
+            //Arrange
+            var expectedAccountName = "katetest.selenium";
+
+            //Act
+            LoginToYandexMail();
+
+            IsElementVisible(By.XPath("//span[text()='katetest.selenium']"));
+            var actualAccountName = this.driver.FindElement(By.XPath("//span[text()='katetest.selenium']")).Text;
```

Comment:
It is unclear which element is taken by XPath. Please define the locator as a separate parameter. For example:
//Arrange
var accountButton = By.XPath("//span[text()='katetest.selenium']");

//Act
var actualAccountName = this.driver.FindElement(accountButton).Text;